File: 0206-reverse-linked-list/0206-reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode *currentNode=head;
        ListNode *previousNode=NULL;
        ListNode *tempNode=NULL;
        
        while(currentNode)
        {
            tempNode=currentNode->next;
            currentNode->next=previousNode;
            previousNode=currentNode;
            currentNode=tempNode;
        }
        
        
        return previousNode;
    }
    
    
};

File: 0206-reverse-linked-list/0206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
         if(head==null || head.next==null)
            return  head;

    ListNode prev=null;
    ListNode curr=head;
    ListNode nextt=head.next;

        while(nextt!=null)
        {
            curr.next=prev;
            prev=curr;
            curr=nextt;
            nextt=nextt.next;
        }   
        curr.next=prev;
        head=curr;

        return head; 
    }
}